### PR TITLE
[MRG] Create plot_subgraphs.py example

### PR DESCRIPTION
**Pull request by lobpcg**
Fixes https://github.com/networkx/networkx/issues/4220

### CODE REVIEW

**Comment by lobpcg** [comment]
LGTM, could someone please review?

**Comment by lobpcg** [comment]
The most recent commit generates the docs at https://2810-890377-gh.circle-artifacts.com/0/doc/build/html/auto_examples/algorithms/plot_subgraphs.html#sphx-glr-auto-examples-algorithms-plot-subgraphs-py 
There appears to be a small deficiency in the plots - some nodes are not plotted for edges removed. It probably is related to the line 81 commented out
```
# nx.draw_networkx_nodes(G_minus_H, _pos, node_color=_node_color_list)
```
Unfortunately, commented in it gives an error, see the previous commit, which I do not know how to fix. I wrote this code 2 years ago... If the reviewer would want this to be fixed, a specific suggestion  would be highly appreciated.

**Comment by lobpcg** [comment]
LGTM, could someone please review? @jarrodmillman @dschult 
The docs for the current commit are at https://2815-890377-gh.circle-artifacts.com/0/doc/build/html/auto_examples/algorithms/plot_subgraphs.html and LGTM

**Review by rossbar — APPROVED** [review]
Generally I think this is a nice example, though it is a bit longer/more involved than most of the other gallery examples. In the longer term it might be worth considering making an "article" out of this example for `nx-guides`, as it may benefit from a bit more narrative text.

That said, IMO there aren't any blockers to including it as a gallery example. There are a lot minor improvements that could be suggested, but I think it might be easier to collate them in a follow-up PR rather than get hung up on nits/suggestions in the discussion here. With that in mind I'm +1 for this - thanks for the nice example @lobpcg !

**Comment by rossbar** [comment]
I took the liberty of pushing up a few rst formatting-related changes to get rid of the sphinx warnings about header length.

File: examples/algorithms/plot_subgraphs.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+G_ex_r = nx.DiGraph()
+# Composing all subgraphs.
+for subgraph in subgraphs_of_G_ex:
+    G_ex_r = nx.compose(G_ex_r, subgraph)
+# Adding the previously stored edges.
+G_ex_r.add_edges_from(removed_edges.edges())
+
+###############################################################################
+# Check that the original grapg and the reconstructed graphs are isomorphic.
+# --------------------------------------------------------------------------
```
Comment (dschult) [inline examples/algorithms/plot_subgraphs.py:189]:
```suggestion
# Check that the original graph and the reconstructed graphs are isomorphic.
# --------------------------------------------------------------------------
```

File: examples/algorithms/plot_subgraphs.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+
+###############################################################################
+# Create an example directed graph.
+# ---------------------------------
+#
+# This directed graph has one input node labeled `in` and plotted in blue color
+# and one output node labeled `out` and plotted in magenta color.
+# The other six nodes are classified as four `supported` plotted in green color
+# and two `unsupported` plotted in red color. The goal is compuiting a list
```
Comment (dschult) [inline examples/algorithms/plot_subgraphs.py:121]:
```suggestion
# and two `unsupported` plotted in red color. The goal is computing a list
```

**Review by dschult — APPROVED** [review]
Looks good to me too.  Thanks @rossbar for finding a good path forward.
And especially Thanks @lobpcg for putting this together!

(two spelling changes listed below)

**Comment by lobpcg** [comment]
This was my exercise for a coding test at a job interview, which I thought would be a waste not to make public. 

I dedicate this work to our cat who passed away today after a long happy life with us.

**Pull request merged**